src/ts-serialize.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import {Class, Function, Interface, Namespace, Property} from './ts-ast';
+
+/**
+ * Encode a TypeScript AST node in TypeScript declaration file syntax (d.ts).
+ */
+export function serializeTsDeclarations(
+    node: Namespace|Class|Interface|Function|Property,
+    depth: number = 0): string {
+  switch (node.kind) {
+    case 'namespace':
+      return serializeNamespace(node, depth);
+    case 'class':
+    case 'interface':
+      return serializeClassOrInterface(node, depth);
+    case 'function':
+      return serializeMethod(node, depth);
+    case 'property':
+      return serializeProperty(node, depth);
```

Comment:
For statically known distinguished types, prefer this pattern for ensuring you got all cases:

```typescript
case default:
  const never: never = node;
  throw new Error(`Unknown kind of node encountered: ${util.inspect(never)}`);
```

This will give you useful runtime and type checking errors if something changes.

Reply:
Cool! Done.